torchdata/stateful_dataloader/stateful_dataloader.py

```diff
@@ -861,9 +868,28 @@ def __init__(self, loader, next_iter_state):
         _utils.signal_handling._set_worker_pids(id(self), tuple(w.pid for w in self._workers))  # type: ignore[misc]
         _utils.signal_handling._set_SIGCHLD_handler()
         self._worker_pids_set = True
-        self._snapshot, self._worker_snapshots, self._main_snapshots = {}, {}, collections.deque()  # type: ignore[var-annotated]
+        self._snapshot, self._main_snapshots = {}, collections.deque()  # type: ignore[var-annotated]
+        self._worker_snapshots: Dict[str, _IncrementalWorkerState] = {}
```

Comment:
AckStartup always sends back an initial state, I think this is needed for this scenario, though code may have moved around since I last visited this:

```
dl = DataLoader(..., persistent_workers=True)
dl.load_state_dict({step: 2})
batches = list(dl)
batches2 = list(dl)
assert len(batches2) > batches
```

Reply:
For comments, I left this in the worker.py, I'll tag you in there. Let me know if you think it would be better here, but at the very least I can add a reference here to worker.py